R/sanitize_models.R

```diff
@@ -0,0 +1,162 @@
+#' Check if a model can be used for predictions for specific data
+#'
+#' This function ensures that a model trained on a given dataset can be used to
+#' make predictions on another specific dataset. It performs two types of
+#' checks: i) level compatibility, and ii) missing data. For i) checks ensure
+#' that factors of the testing set do not contain any new levels. For ii), it
+#' ensures that there are no NAs in predictors of the training or testing sets.
+#'
+#' @author Thibaut Jombart
+#'
+#' @noRd
+#'
+#' @param model A [`trending_model()`] object.
+#'
+#' @param x_training A `data.frame` used as training set.
+#' 
+#' @param x_testing A `data.frame` used as testing set.
+#'
+#' @param warn A `logical` indicating if warnings should be generated issues are
+#'     found.
+#'
+#' @return A single logical indicating if all checks were OK (TRUE); FALSE if
+#'     there was a single issue, either due to incompatible levels or missing
+#'     data.
+
+sanitize_model <- function(model,
+                           x_training,
+                           x_testing,
+                           warn = FALSE) {
+    
+  # Auxiliary functions
+  
+  ## Check level consistency for a single factor
+  ##
+  ## Returns TRUE if levels are okay, FALSE otherwise
+  check_factor <- function(f_train, f_test) {
```

Comment:
Is there a reason to define this within sanitize_model?  This means it gets defined each time the function is called.

Reply:
Will move to separate files